my-app/src/app/modules/courses-page/services/courses-handler.service.ts

```diff
@@ -0,0 +1,39 @@
+import {Injectable} from '@angular/core';
+
+import {COURSE_DATA} from "@assets/mocks/course-data.mock";
+
+import {ICourse} from "@shared/";
+
+@Injectable({
+  providedIn: 'root'
+})
+
+export class CoursesHandlerService {
+  private initialCourses: ICourse[] = COURSE_DATA;
+  private visibleCourses: ICourse[] = structuredClone(this.initialCourses);
+
+  constructor() {
+  }
+
+  public getList(): ICourse[] {
+    return this.visibleCourses;
+  }
+
+  // public createCourse(properties: Course, arr: Course[]): Course[] {
+  //   arr.push(properties);
+  //   return arr;
+  // }
+  //
+  public getItemById(id: number): ICourse | undefined {
+    return this.visibleCourses.find(item => item.id === id);
+  }
+
+  //
+  // public updateItem() {
+  // }
+
+  public removeItem(id: number): ICourse[] {
+    this.visibleCourses = this.visibleCourses.filter(item => item.id !== id);
+    return this.visibleCourses
```

Comment:
There's a good principle named CQRS. No need to change here something right now, just read about it